plutus-ledger-api/test/Spec/Interval.hs

```diff
@@ -6,15 +6,46 @@
 module Spec.Interval where
 
 import Data.List (sort)
+import Data.Maybe (fromJust)
+import Data.Set qualified as Set
 import Hedgehog (Property, forAll, property)
 import Hedgehog qualified
 import Hedgehog.Gen qualified as Gen
+import Hedgehog.Laws.Eq
+import Hedgehog.Laws.Lattice
+import Hedgehog.Laws.Ord
 import Hedgehog.Range qualified as Range
 import PlutusLedgerApi.V1.Interval qualified as Interval
+import PlutusPrelude (reoption)
 import Test.Tasty (TestTree, testGroup)
-import Test.Tasty.Hedgehog (testPropertyNamed)
+import Test.Tasty.Hedgehog (testProperty)
 import Test.Tasty.HUnit (assertBool, testCase)
 
+-- TODO: maybe bias towards generating non-empty intervals?
+
+genExtended :: Bool -> Hedgehog.Gen a -> Hedgehog.Gen (Interval.Extended a)
+genExtended finite g =
+  if finite
+  then Interval.Finite <$> g
+  else Gen.choice [ Interval.Finite <$> g, pure Interval.NegInf, pure Interval.PosInf ]
+
+genLowerBound :: Bool -> Hedgehog.Gen a -> Hedgehog.Gen (Interval.LowerBound a)
+genLowerBound finite g = Interval.LowerBound <$> genExtended finite g <*> Gen.bool
+
+genUpperBound :: Bool -> Hedgehog.Gen a -> Hedgehog.Gen (Interval.UpperBound a)
+genUpperBound finite g = Interval.UpperBound <$> genExtended finite g <*> Gen.bool
+
+genInterval :: Bool -> Hedgehog.Gen a -> Hedgehog.Gen (Interval.Interval a)
+genInterval finite g = Interval.Interval <$> genLowerBound finite g <*> genUpperBound finite g
+
+genFiniteDiscreteInterval :: Hedgehog.Gen (Interval.Interval Integer)
+genFiniteDiscreteInterval = genInterval True (Gen.integral (toInteger <$> Range.linear @Int 0 100))
+
+genDiscreteInterval :: Hedgehog.Gen (Interval.Interval Integer)
+genDiscreteInterval = genInterval False (Gen.integral (toInteger <$> Range.linear @Int 0 100))
```

Comment:
How about generating other staff than Integer, e.g. Bool?

Reply:
I added some tests for closed boolean intervals, open ones obviously don't work.